cpp/oneapi/dal/test/engine/metrics/clustering.hpp

```diff
@@ -0,0 +1,86 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include <iostream>
+
+#include "oneapi/dal/test/engine/common.hpp"
+
+#include "oneapi/dal/table/row_accessor.hpp"
+#include "oneapi/dal/table/homogen.hpp"
+#include "oneapi/dal/table/detail/table_builder.hpp"
+
+namespace de = oneapi::dal::detail;
+
+namespace oneapi::dal::test::engine {
+
+template <typename Float = double>
+auto davies_bouldin_index(const table& data, const table& centroids, const table& assignments) {
```

Comment:
Suggest to always declare type if it does not involve very complex type inference logic or is not obvious. This function doesn't use heavy templates and is not obvious.